packages/components/src/components/date-picker/date-picker-native.jsx

```diff
@@ -85,7 +85,17 @@ const Native = ({
                 onFocus={handleFocus}
                 disabled={disabled}
                 onChange={e => {
-                    onSelect(e.target.value);
+                    let new_value = e.target.value;
+
+                    if (max_date) {
+                        const moment_maxdate = toMoment(max_date);
+                        const moment_value = toMoment(new_value);
+                        const days_diff = moment_maxdate.diff(moment_value, 'days');
+
+                        new_value = days_diff < 0 ? moment_maxdate.format('YYYY-MM-DD') : new_value;
+                    }
+
+                    onSelect(new_value);
```

Comment:
@prince-deriv With this approach, the `value` of the input datepicker element won't be same. 
Once we selected the future date and when we reopen the datepicker, the datepicker will still have the future date selected, but in the UI we will show the `max_date`.

Also, we should use similar kind of logic to reset to `min_date` when the date selected is before the `min_date`

Reply:
will check the input value and ui value and also add the same logic in `min_date`